src/lsp4clj/server.clj

```diff
@@ -96,12 +90,17 @@
   Sends `$/cancelRequest` only once, though `lsp4clj.server/deref-or-cancel` or
   `future-cancel` can be called multiple times."
   [id method started server]
-  (map->PendingRequest {:p (promise)
-                        :cancelled? (atom false)
-                        :id id
-                        :method method
-                        :started started
-                        :server server}))
+  ;; Chaining `(-> (p/deferred) (p/catch ...))` seems like it should work, but
+  ;; doesn't.
```

Comment:
I think this comment is obsolete, as we identified why we cannot chain: We need to cancel `p`, not the wrapped promise returned by `p/catch`.